### ISSUE TITLE
Support local files and protected pages

### ISSUE THREAD

**Issue by mkoushan** [ThomazPom/Moz-Ext-UltimaDark#105]
### DO NOT CREATE UNIQUE ISSUES FOR FEATURE REQUESTS

- [X] I understand that I should not be creating individual issues for feature requests or submitting this template after reading it

**Comment by Vintagemotors** [comment]
I assumed this worked with permissions allowed but you are correct that it does not (by default at least) as it is blocked by Firefox's security policy. You can try changing  ``extensions.quarantinedDomains.enabled`` to false in ``about:config`` (made no difference in my testing) but I see people have the same issue with other extensions so it's not unique to UltimaDark.  I do not see an appropriate permission available to correct this either (unless the "storage" permission grants access to the user's entire file system and not just the folder allocated by Firefox for the extension to use). [docs](https://extensionworkshop.com/documentation/develop/request-the-right-permissions/)

**Comment by mkoushan** [comment]
> I assumed this worked with permissions allowed but you are correct that it does not (by default at least) as it is blocked by Firefox's security policy. You can try changing `extensions.quarantinedDomains.enabled` to false in `about:config` (made no difference in my testing) but I see people have the same issue with other extensions so it's not unique to UltimaDark. I do not see an appropriate permission available to correct this either (unless the "storage" permission grants access to the user's entire file system and not just the folder allocated by Firefox for the extension to use). [docs](https://extensionworkshop.com/documentation/develop/request-the-right-permissions/)

I have that settings disabled from Dark Reader extension. That time it darken my local html files (I've downloaded cpp reference). Dark Reader makes it dark because it changes the css after the page gets loaded. But in Ultima you change it in the streamline before getting to gecko and I think that's why it don't work with local file. I guess the streamline with local html files are different, if I get time I'll look into it currently I'm really busy doing life.

**Comment by mkoushan** [comment]
With or Without Ultima:
![image](https://github.com/user-attachments/assets/f748e7f9-23a0-46ec-98d5-f698ebb56c02)

With Dark Reader:
![image](https://github.com/user-attachments/assets/64ba61f9-823d-4c6a-8e7d-400a20ea04db)

**Comment by atomGit** [comment]
late to the party, but i would [not recommend](https://support.mozilla.org/en-US/kb/quarantined-domains) disabling `extensions.quarantinedDomains.enabled` and if an extension requires that it be disabled, then you might consider discarding it

**Comment by ThomazPom** [comment]
This is true. I would not allow any extension to access these sensitive domains on my browser instance. It would enable extensions to act on my behalf, potentially doing anything. For example, an extension could rate itself or upload other extensions to AMO. This is not problematic with well-known extensions, but who knows what malicious code might be hidden in small, unpopular ones.

[Mozilla Support: Quarantined Domains](https://support.mozilla.org/en-US/kb/quarantined-domains)

**Comment by atomGit** [comment]
> This is not problematic with well-known extensions, but who knows what malicious code might be hidden in small, unpopular ones.

actually the opposite can be the case - from the very nice and ethical dev of the once super-popular All-in-One Sidebar...

> "It was always very important for me to be honest and fair to the users. I had very good offers to sell the extension, but I didn't want to see that AiOS turn into adware or spyware. "

that said, as of about 3-4 months ago i'm seeing a *flood* of new add-ons on AMO that are very sketchy looking and share several commonalities...

* the dev joined AMO the same day the add-on is published
* the dev has only 1 add-on
* the add-ons are very simple
* the add-ons usually, but maybe not always, use the default puzzle piece icon
* the manifest.json is often incomplete or doesn't match the purpose of the add-on, as though it was copied from elsewhere
* these add-ons appear consecutively in batches - you can see this if you sort by 'Recently Updated' - i see new batches every day
* no links to a code repo or support
* the 'Related Categories' section of the AMO page is unrelated to the stated purpose of the add-on
* the add-ons have 0 users (so even the dev didn't install it from AMO)
* the add-ons have 0 reviews, so even the dev didn't give it 5 stars as they often do
* the names used by the dev(s) sometimes follow a similar theme per batch

i'm not a JS coder but when i look at the code for these add-ons, while they don't appear malicious, they also may not perform the advertised task either, sometimes just loading a URL that may be malicious, or not

one possibility might be that they're waiting for one or more of these add-ons to become popular at which point they'll "update" it, but that's just a guess
